test/groovy/UpdatecliStepTests.groovy

```diff
@@ -136,4 +141,32 @@ class UpdatecliStepTests extends BaseTest {
     // And the custom credentialsId is taken in account
     assertTrue(assertMethodCallContainsPattern('usernamePassword', "credentialsId=${anotherCredentialsId}"))
   }
+
+  /**
+   * New Test Cases for Custom Version Functionality
+   */
+
+  // Test that when a custom version is specified, the pipeline includes the download steps.
+  @Test
+  void itRunSuccessfullyWithCustomVersion() throws Exception {
+    def script = loadScript(scriptName)
+    script.call(version: '0.92.0')
+    printCallStack()
+    assertJobStatusSuccess()
+    assertTrue(assertMethodCallContainsPattern('sh', 'curl --silent --show-error --location --output'))
+    assertTrue(assertMethodCallContainsPattern('sh', 'mkdir -p "${CUSTOM_UPDATECLI_PATH}"'))
+    assertTrue(assertMethodCallContainsPattern('sh', 'tar --extract --gzip --file="${tarFileName}" --directory="${CUSTOM_UPDATECLI_PATH}" updatecli'))
+    assertTrue(assertMethodCallContainsPattern('sh', 'updatecli diff'))
+  }
+
+  @Test
+  void itFailsWhenCustomVersionNotFound() throws Exception {
+    def script = loadScript(scriptName)
+    try {
+      script.call(version: '0.99.99')
+      fail("Expected error due to custom version not found")
+    } catch (Exception e) {
+      // Test passes if an exception is thrown due to curl failure.
+    }
+  }
```

Comment:
This test seems to test nothing (it will always pass). I remember a version of it in which you where simulating a failure of the `sh` step running `curl`: did it got lost?

We can also remove this test and get done with it: your choice